src/bulk_insert/bulk_insert.c

```diff
@@ -232,59 +244,68 @@ int BulkInsert(RedisModuleCtx *ctx, GraphContext *gc, RedisModuleString **argv,
 		return BULK_FAIL;
 	}
 
-	// Read the number of node tokens
+	// read the number of node tokens
 	long long node_token_count;
 	long long relation_token_count;
+
 	if(RedisModule_StringToLongLong(*argv++, &node_token_count)  != REDISMODULE_OK) {
 		RedisModule_ReplyWithError(ctx, "Error parsing number of node descriptor tokens.");
 		return BULK_FAIL;
 	}
 
-	// Read the number of relation tokens
+	// read the number of relation tokens
 	if(RedisModule_StringToLongLong(*argv++, &relation_token_count)  != REDISMODULE_OK) {
 		RedisModule_ReplyWithError(ctx, "Error parsing number of relation descriptor tokens.");
 		return BULK_FAIL;
 	}
-	argc -= 2;
 
+	argc -= 2;
 	PendingInserts bulk_ctx = { .gc = gc };
 
-	// Stack-allocate an array to contain the attribute key IDs of each label file.
+	// stack-allocate an array to contain the attribute key IDs
+	// of each label file
+	//
+	// TODO: not sure how many node tokens to expect, but for large numbers
+	// allocating `props_per_label` on the stack will fail
 	Attribute_ID *props_per_label[node_token_count];
 	if(node_token_count > 0) {
 		bulk_ctx.nodes = rm_malloc(node_count * sizeof(Node));
 		bulk_ctx.node_props = rm_malloc(node_count * sizeof(PendingProperties));
-		// Process all node files
-		int rc = _BulkInsert_ProcessTokens(&bulk_ctx, node_token_count, &argv, &argc,
-										   props_per_label, SCHEMA_NODE);
+		// process all node files
+		int rc = _BulkInsert_ProcessTokens(&bulk_ctx, node_token_count, &argv,
+				&argc, props_per_label, SCHEMA_NODE);
 		if(rc != BULK_OK) return BULK_FAIL;
 	}
 
-	// Stack-allocate an array to contain the attribute key IDs of each relationship type file.
+	// stack-allocate an array to contain the attribute key IDs
+	// of each relationship type file
+	//
+	// TODO: not sure how many node tokens to expect, but for large numbers
```

Comment:
We have one node token for up to 500 megabytes of each label/reltype, so I'm not worried about stack smashing here.